front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -45,9 +51,94 @@ const ManchetteWithSpaceTimeChartWrapper = ({
 
   const [waypointsPanelIsOpen, setWaypointsPanelIsOpen] = useState(false);
 
+  // Cut the space time chart curves if the first or last waypoints are hidden
+  const spaceTimeChartLayersData = useMemo(() => {
+    let filteredProjectPathTrainResult = projectPathTrainResult;
+    let filteredConflicts = conflicts;
+
+    if (!waypointsPanelData) return { filteredProjectPathTrainResult, filteredConflicts };
+
+    const { filteredWaypoints } = waypointsPanelData;
+    const firstPositon = filteredWaypoints.at(0)!.position;
+    const lastPosition = filteredWaypoints.at(-1)!.position;
+
+    if (
+      filteredWaypoints.length >= 2 &&
+      (firstPositon !== 0 || lastPosition !== operationalPoints.at(-1)!.position)
+    ) {
+      filteredProjectPathTrainResult = projectPathTrainResult.map((train) => ({
+        ...train,
+        space_time_curves: train.space_time_curves.map(({ positions, times }) => {
+          const cutPositions = new Set<number>();
+          const cutTimes = new Set<number>();
+
+          for (let i = 1; i < positions.length; i += 1) {
+            const currentRange: LayerRangeData = {
+              spaceStart: positions[i - 1],
+              spaceEnd: positions[i],
+              timeStart: times[i - 1],
+              timeEnd: times[i],
+            };
+
+            const interpolatedRange = interpolateRange(currentRange, firstPositon, lastPosition);
+
+            // TODO : remove reformatting the datas when https://github.com/OpenRailAssociation/osrd-ui/issues/694 is merged
+            if (interpolatedRange) {
+              cutPositions.add(interpolatedRange.spaceStart);
+              cutPositions.add(interpolatedRange.spaceEnd);
+              cutTimes.add(interpolatedRange.timeStart);
+              cutTimes.add(interpolatedRange.timeEnd);
+            }
+          }
+
+          return {
+            positions: Array.from(cutPositions),
+            times: Array.from(cutTimes),
+          };
+        }),
+        signal_updates: compact(
+          train.signal_updates.map((signal) => {
+            const formattedSignal = {
+              ...omit(signal, ['position_start', 'position_end', 'time_start', 'time_end']),
+              spaceStart: signal.position_start,
+              spaceEnd: signal.position_end,
+              timeStart: signal.time_start,
+              timeEnd: signal.time_end,
+            };
+            const { spaceStart, spaceEnd, timeStart, timeEnd } = formattedSignal;
+            const updatedSignalRange = interpolateRange(
+              { spaceStart, spaceEnd, timeStart, timeEnd },
+              firstPositon,
+              lastPosition
+            );
```

Comment:
You can directly call the function interpolateRange
```suggestion
            const updatedSignalRange = interpolateRange(
              {
                spaceStart: signal.position_start,
                spaceEnd: signal.position_end,
                timeStart: signal.time_start,
                timeEnd: signal.time_end,
              },
              firstPositon,
              lastPosition
            );
```

Reply:
Done